agent/src/main/java/org/openremote/agent/protocol/dmx/artnet/ArtnetClientProtocol.java

```diff
@@ -0,0 +1,502 @@
+                    put("amountOfLeds", Values.create(3));
+                }})));
+            }})
+    );
+    public static final List<MetaItemDescriptor> ATTRIBUTE_META_ITEM_DESCRIPTORS = Arrays.asList(
+            META_ATTRIBUTE_MATCH_FILTERS,
+            META_ATTRIBUTE_MATCH_PREDICATE,
+            META_ARTNET_LIGHT_ID,
+            META_ARTNET_CONFIGURATION);
+
+    protected final Map<AttributeRef, List<Pair<AttributeRef, Consumer<ArtnetPacket>>>> protocolMessageConsumers = new HashMap<>();
+
+    private List<AbstractArtnetLight> artnetLightMemory = new ArrayList<>();
+
+    @Override
+    public String getProtocolName() {
+        return PROTOCOL_NAME;
+    }
+
+    @Override
+    public String getProtocolDisplayName() {
+        return PROTOCOL_DISPLAY_NAME;
+    }
+
+    @Override
+    public String getVersion() {
+        return PROTOCOL_VERSION;
+    }
+
+    @Override
+    public List<AbstractArtnetLight> getLightMemory() { return artnetLightMemory; }
+
+    @Override
+    protected List<MetaItemDescriptor> getProtocolConfigurationMetaItemDescriptors() {
+        return PROTOCOL_META_ITEM_DESCRIPTORS;
+    }
+
+    @Override
+    protected List<MetaItemDescriptor> getLinkedAttributeMetaItemDescriptors() {
+        return ATTRIBUTE_META_ITEM_DESCRIPTORS;
+    }
+
+    @Override
+    public AssetAttribute getProtocolConfigurationTemplate() {
+        return super.getProtocolConfigurationTemplate()
+                .addMeta(
+                        new MetaItem(META_PROTOCOL_HOST, null),
+                        new MetaItem(META_PROTOCOL_PORT, null)
+                );
+    }
+
+    @Override
+    protected void doUnlinkProtocolConfiguration(AssetAttribute protocolConfiguration) {
+        synchronized (protocolMessageConsumers) {
+            protocolMessageConsumers.remove(protocolConfiguration.getReferenceOrThrow());
+        }
+        super.doUnlinkProtocolConfiguration(protocolConfiguration);
+    }
+
+    @Override
+    protected Supplier<ChannelHandler[]> getEncoderDecoderProvider(UdpIoClient<ArtnetPacket> client, AssetAttribute protocolConfiguration) {
+        Supplier<ChannelHandler[]> encoderDecoderProvider = () -> {
+            List<ChannelHandler> encodersDecoders = new ArrayList<>();
+            encodersDecoders.add(new AbstractNettyIoClient.MessageToByteEncoder<ArtnetPacket>(ArtnetPacket.class, client, new BiConsumer<ArtnetPacket, ByteBuf>() {
+                @Override
+                public void accept(ArtnetPacket packet, ByteBuf buf) {
+                    packet.toByteBuf(buf);
+                }
+            }));
+            return encodersDecoders.toArray(new ChannelHandler[0]);
+        };
+        return encoderDecoderProvider;
+    }
+
+    @Override
+    protected void doLinkAttribute(AssetAttribute attribute, AssetAttribute protocolConfiguration) {
+        AttributeRef protocolRef = protocolConfiguration.getReferenceOrThrow();
+        Consumer<ArtnetPacket> messageConsumer = artnetPacket -> {};
```

Comment:
What's the purpose of this empty consumer? How are linked attributes updated when a change occurs to a light for example?